packages/json-schema/test/integration/build-schema.test.ts

```diff
@@ -0,0 +1,216 @@
+// Copyright IBM Corp. 2013,2018. All Rights Reserved.
+// Node module: @loopback/json-schema
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {model, property, ModelMetadataHelper} from '@loopback/repository';
+import {modelToJsonDef, toJsonProperty} from '../../src/build-schema';
+import {expect} from '@loopback/testlab';
+
+describe('build-schema', () => {
+  context('exception', () => {
+    it('errors out when "@property.array" is not used on an array', () => {
+      @model()
+      class BadArray {
+        @property() badArr: string[];
+      }
+      expect(() => {
+        modelToJsonDef(BadArray);
+      }).to.throw(/type is defined as an array/);
+    });
+  });
+  describe('modelToJSON', () => {
+    it('does not convert null or undefined property', () => {
+      @model()
+      class TestModel {
+        @property() nul: null;
+        @property() undef: undefined;
+      }
+      const jsonDef = modelToJsonDef(TestModel);
+      expect(jsonDef.properties).to.not.containEql({
+        nul: {type: 'null'},
```

Comment:
I find this check weird and possible brittle. In theory, there is an infinite number of ways how to convert a null property into JSON schema, almost all of them invalid, and here we are ruling out only one of the invalid options.

For example, the following output would pass this test:

```js
{
  properties: {
    nul: { anyOf: [{type: 'null'}] }
  }
}
```

I think you wanted to assert that `jsonDef.properties` does not contain any entry for the `nul` property, right? 

```js
expect(jsonDef.properties).to.not.have.key('nul');
```

The same comment applies to next check for `undef` below.